File: 프로그래머스/0/120890. 가까운 수/가까운 수.java
class Solution {
    public int solution(int[] array, int n) {
        int answer = array[0];
        int minDiff = Math.abs(n - answer);
        
        for (int num : array) {
            int diff = Math.abs(n - num);
            if (diff < minDiff || (diff == minDiff && num < answer)) {
                minDiff = diff;
                answer = num;
            }
        }
        return answer;
    }
}

File: 프로그래머스/0/120890. 가까운 수/가까운 수.js
function solution(array, n) {
  return array.reduce((acc, cur) => {
    const accDiff = Math.abs(n - acc);
    const curDiff = Math.abs(n - cur);
    return curDiff < accDiff || (curDiff === accDiff && cur < acc) ? cur : acc;
  });
}
